### PR TITLE
`dunfell` update v2024.06

### PR DESCRIPTION
**Pull request by lluiscampos**
The full update. It includes #2121.

### CODE REVIEW

File: meta-mender-commercial/recipes-mender/mender-binary-delta/mender-binary-delta.inc [hunk 1]
```diff
@@ -36,14 +49,22 @@ do_version_check() {
 addtask do_version_check after do_patch before do_install
 
 do_configure() {
-    # We need the sed operation because bitbake misinterprets the JSON closing
-    # brace as end of function, unless we put spaces in front of it.
-    sed -e 's/^ //' <<EOF | cat > ${WORKDIR}/mender-binary-delta.conf
- {
+    echo "{" > ${WORKDIR}/mender-binary-delta.conf
+
+    if [ -n "${MENDER_FLASH_TOOL}" ]; then
```
Comment (kacf) [inline meta-mender-commercial/recipes-mender/mender-binary-delta/mender-binary-delta.inc:54]: Oh darn! I just realized this is wrong in both branches (master too)!

It needs to check the value, and set either `mender-flash` or `ubiupdatevol`, not just `mender-flash` unconditionally!

Could you fix both?

File: meta-mender-core/classes/mender-setup.bbclass [hunk 2]
```diff
@@ -244,10 +244,10 @@ python() {
 addhandler mender_sanity_handler
 mender_sanity_handler[eventmask] = "bb.event.ParseCompleted"
 python mender_sanity_handler() {
-    if bb.utils.contains('MENDER_FEATURES_ENABLE', 'mender-partuuid', True, False, d) and d.getVar('MENDER_STORAGE_DEVICE', True) != "":
+    if bb.utils.contains('MENDER_FEATURES', 'mender-partuuid', True, False, d) and d.getVar('MENDER_STORAGE_DEVICE', True) != "":
         bb.warn("MENDER_STORAGE_DEVICE is ignored when mender-partuuid is enabled. Clear MENDER_STORAGE_DEVICE to remove this warning.")
 
-    if bb.utils.contains('MENDER_FEATURES_ENABLE', 'mender-partuuid', True, False, d) and bb.utils.contains('MENDER_FEATURES_ENABLE', 'mender-uboot', True, False, d):
+    if bb.utils.contains('MENDER_FEATURES', 'mender-partuuid', True, False, d) and bb.utils.contains('MENDER_FEATURES', 'mender-uboot', True, False, d):
```
Comment (kacf) [inline meta-mender-core/classes/mender-setup.bbclass:250]: Careful! `MENDER_FEATURES_ENABLE` end up in `DISTRO_FEATURES` in dunfell and older. From grepping I can see that there are already a couple of mistakes in dunfell regarding this, notably when handling `mender-partuuid`.

**Review by kacf — APPROVED** [review]
Pre-approving, I don't think it needs re-reviewing. But there are a couple things to fix below.

File: meta-mender-commercial/recipes-mender/mender-binary-delta/mender-binary-delta.inc (hunk 1 above)
Comment (lluiscampos) [reply meta-mender-commercial/recipes-mender/mender-binary-delta/mender-binary-delta.inc:54]: Cherry-picked now :+1:

File: meta-mender-core/classes/mender-setup.bbclass (hunk 2 above)
Comment (lluiscampos) [reply meta-mender-core/classes/mender-setup.bbclass]: Thanks.

Fixed this on the cherry-pick and also added a [separate commit for the existing broken feature](https://github.com/mendersoftware/meta-mender/pull/2125/commits/95f38e932a3e44b44b9d357f705ac087d617a765). Can you please give feedback on that commit? Is the changelog too much and we could do `Title` instead?

Comment (kacf) [reply meta-mender-core/classes/mender-setup.bbclass]: I think I'd go for "title", the full message is a bit much IMHO. Looks good otherwise!

**Comment by lluiscampos** [comment]
The last force-push only changed a commit message. The previous pipeline (no code changes) can be found [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/pipelines/1320969046). Merging :rocket:

**Pull request merged**